### PR TITLE
Personal Power Regeneration added to _onRest

### PR DESCRIPTION
**Pull request by JFranklin3000**
##### In this PR
- Added Personal Power Regeneration to _onRest
- 

##### Testing
- pack Power Rangers Core Rulebook
- Drop Role on to Actor
- Check Personal Power Regeneration on Role
- click rest button. 
- validate proper amount of Personal Power Restored.

fvtt package unpack -n "prcrbitems"

### CODE REVIEW

File: module/sheets/actor-sheet.mjs [hunk 1]
```diff
@@ -472,6 +472,17 @@ export class Essence20ActorSheet extends ActorSheet {
       ui.notifications.info(`Energon restored by ${energonRestore}.`);
     }
 
+    //Reseting Personal Power
+    let powerRestore = 0;
+    if (this.actor.system.powers.personal.max > 0) {
+      powerRestore = this.actor.system.powers.personal.value + this.actor.system.powers.personal.regeneration;
+      if (powerRestore > this.actor.system.powers.personal.max) {
+        powerRestore = this.actor.system.powers.personal.max;
+      }
```
Comment (phildominguez) [inline module/sheets/actor-sheet.mjs:481]: We can use Math.min instead here

File: module/sheets/actor-sheet.mjs [hunk 2]
```diff
@@ -472,6 +472,17 @@ export class Essence20ActorSheet extends ActorSheet {
...
+
+      ui.notifications.info("Personal Power restored.");
```
Comment (phildominguez) [inline module/sheets/actor-sheet.mjs:483]: Gah, we missed a bunch of localizations in this method. Want to take care of those?

File: module/helpers/utils.mjs [hunk 3]
```diff
@@ -520,6 +520,7 @@ export async function setRoleValues(role, actor, newLevel=null, previousLevel=nu
 
     await actor.update({
       "system.powers.personal.max": newPersonalPowerMax,
```
Comment (phildominguez) [inline module/helpers/utils.mjs:522]: Nothing you did here, but leveling is setting max to 0. Let's get the fix in this PR.

**Review by phildominguez — CHANGES_REQUESTED** [review]

File: module/helpers/utils.mjs (hunk 3 above)
Comment (JFranklin3000) [reply module/helpers/utils.mjs:522]: fixed

File: module/sheets/actor-sheet.mjs (hunk 2 above)
Comment (JFranklin3000) [reply module/sheets/actor-sheet.mjs:483]: Got them all

File: module/sheets/actor-sheet.mjs (hunk 1 above)
Comment (JFranklin3000) [reply module/sheets/actor-sheet.mjs:481]: changed

**Review by phildominguez — APPROVED** [review]

**Pull request merged**